CRM/Core/Form/Tag.php

```diff
@@ -19,7 +19,6 @@
  * This class generates form element for free tag widget.
  */
 class CRM_Core_Form_Tag {
-  public $_entityTagValues;
```

Comment:
Do you need to add this to CRM_Core_Form? Since $form lower down where it's used could be any form?

Reply:
I agree `_entityTagValues` either needs to be on `CRM_Core_Form` or on every individual form that uses `CRM_Core_Form_Tag`.

I worry a bit about `CRM_Core_Form` becoming super bloated with properties. In cases like this we probably want to make sure to document in which class they'll actually be used.

Reply:
Yeah - generally I like to move stuff like that to a function rather than a property - which can use `Civi::statics` - rather than a form property (but would need to be on a trait I guess)